src/CommandBasedBackendRegisterInfo.cc

```diff
@@ -13,18 +106,180 @@ namespace ChimeraTK {
     writeResponsePattern(writeResponsePattern_), readCommandPattern(readCommandPattern_),
     readResponsePattern(readResponsePattern_), nLinesReadResponse(nLinesReadResponse_), internalType(type),
     delimiter(delimiter_) {
-    if(internalType == InternalType::INT64) {
-      dataDescriptor = DataDescriptor(DataType::int64);
+    // set dataDescriptor from type.
+    switch(type) {
+      case InternalType::UINT64:
+        dataDescriptor = DataDescriptor(
+            DataDescriptor::FundamentalType::numeric, /*isIntegral*/ true, /*isSigned*/ false, /*nDigits*/ 11);
+        break;
+      case InternalType::STRING:
+        dataDescriptor = DataDescriptor(DataDescriptor::FundamentalType::string);
+        break;
+      case InternalType::DOUBLE:
+        // smallest possible 5e-324, largest 2e308
+        dataDescriptor = DataDescriptor(DataDescriptor::FundamentalType::numeric, /*isIntegral*/ false,
+            /*isSigned*/ true, /*nDigits*/ 3 + 325, /*nFragtionalDigits*/ 325);
+        break;
+      case InternalType::VOID:
+        dataDescriptor = DataDescriptor(DataDescriptor::FundamentalType::nodata);
+        break;
+      default:
+      case InternalType::INT64:
+        dataDescriptor = DataDescriptor(
+            DataDescriptor::FundamentalType::numeric, /*isIntegral*/ true, /*isSigned*/ true, /*nDigits*/ 11);
+    }
+  } // end constructor
+
+  /******************************************************************************************************************/
+
+  CommandBasedBackend::MapFile CommandBasedBackend::parse(const std::string& mapFileName) {
+    std::ifstream file(mapFileName);
+    if(!file.is_open()) {
+      throw ChimeraTK::logic_error("Could not open the map file " + mapFileName);
+    }
+
+    json j;
+    file >> j;
+    return j.get<CommandBasedBackend::MapFile>();
+    // get() calls from_json(const json& j, CommandBasedBackend::MapFile& d)
+  }
+
+  /********************************************************************************************************************/
+
+  /**
+   * Throws a ChimeraTK::logic_error if the json key is not in an std::array of valid keys
+   */
+  template<size_t N>
+  void throwIfHasInvalidJsonKey(const json& j, const std::array<std::string, N>& validKeys, std::string errorMessage) {
+    for(auto it = j.begin(); it != j.end(); ++it) {
+      bool stringIsNotInArray = true;
+      for(const std::string& i : validKeys) {
+        if(it.key() == i) {
+          stringIsNotInArray = false;
+          break;
+        }
+      }
+
+      if(stringIsNotInArray) {
+        throw ChimeraTK::logic_error(errorMessage + " " + it.key());
+      }
+    }
+  }
+
+  /********************************************************************************************************************/
+
+  void from_json(const json& j, CommandBasedBackend::MapFile& d) {
```

Comment:
If I look at the content of from_json() I don't see the point of using this "it's found and used by json::get()" feature. I don't like the signature of getting an object that is modified instead of returned, and the content would not change much if this was part of the backend's parse() function (though one might have multiple functions to structure the code).